### DISCUSSION TITLE
i18n locale as URL suffix instead of prefix (using sub-path routing)

### DISCUSSION THREAD

**Discussion by Deckstar** [vercel/next.js#23419] [Ideas]
### Describe the feature you'd like to request

NextJS currently makes it very easy to implement locales with sub-path routing, e.g. (as seen in the documentation):
```
domain.com/blog
domain.com/fr/blog
domain.com/nl-nl/blog
```

However, it would be great if the locale could also be configured as a suffix to the URL, i.e. showing up at the end of the URL instead of the beginning:
```
domain.com/fr
domain.com/blog/fr
domain.com/blog/post/[:id]/fr
```

### Describe the solution you'd like

Ideally it would be great to have this as a one-liner in the `next.config.js`:
```js
module.exports = {
  i18n: {
    defaultLocale: "en",
    locales: ["en", "fr"],
    suffix-locale: true  // <------ this, with default: false, of course
    // ...rest
  },
};

```

### Describe alternatives you've considered

So far we've considered creating a custom Next.js server, but obviously this is not recommending and might end with us having more problems than it's worth.

The reason we wish to have the locales at the end is because we are rebuilding a client's website with Next.JS, and they requested us to keep the links as close to the original website as possible, because they are worried about losing their SEO indexing & ranking for their hundreds of pages. The old website was made with PHP and had the locales at the end.

This feature request can also be a question: is there perhaps some easy, non-obvious way to do this? 🙂

Thank you very much for your time!

**Comment by ravitejaobelisk** [comment]
@Deckstar Looking for a similar setting for one of my project, did you find any workaround or solution?

**Reply by Deckstar** [reply]
> @Deckstar Looking for a similar setting for one of my project, did you find any workaround or solution? 

Unfortunately not 😔 we went with an awful solution where every single link had to have an href and an as prop. Then we wrote all the link paths in Next redirects in the config and disabled file system routing.

This solution was really horrendous and cost us innumerable hours of development time, but we had to go with it because the client insisted. I would strongly recommend to just go with the default locale routing if you can. If there's an SEO issue (e.g. old site had this links format) then you can probably add some SEO meta tags in the head element that describe the old URL, and add redirects from the old format to the new format.

But if you're still forced to do the same, make sure you make a lot of until functions that will create the links for you (in-line interpolated strings will lead to a lot of mistakes down the line 😄).
